### PR TITLE
feat(Log): add basic support for Log domain, update Runtime types

### PR DESCRIPTION
**Pull request by RyuuGan**
And log Domain basic support

Also file log is called `logs` which I don't like because module `log` is already used.

### CODE REVIEW

File: src/protocol/logs.rs [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    #[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
+    #[serde(rename_all = "camelCase")]
+    pub struct LogEntry {
+        pub source: LogEntrySource,
+        pub level: LogEntryLevel,
+        pub text: String,
+        /// Timestamp when this entry was added
+        /// Number of milliseconds since epoch
+        /// See https://chromedevtools.github.io/devtools-protocol/tot/Runtime#type-Timestamp
+        pub timestamp: f64,
```
Comment (lukaslueg) [inline src/protocol/logs.rs]: Maybe we want something from `chrono` here so there is no ambiguity.

File: src/protocol/runtime.rs [hunk 2]
```diff
@@ -25,19 +26,90 @@ pub mod methods {
...
+
+    /// Call frames for assertions or error messages
+    /// See https://chromedevtools.github.io/devtools-protocol/tot/Runtime#type-StackTrace
+    #[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+    #[serde(rename_all = "camelCase")]
+    pub struct StackTrace {
+        description: Option<String>,
+        call_frames: Vec<CallFrame>,
+        // parent: Option<StackTrace>,
+        /// Experimental
```
Comment (lukaslueg) [inline src/protocol/runtime.rs]: Could you expand on the comment, giving some words about what is experimental and why.

Comment (RyuuGan) [reply src/protocol/runtime.rs]: Yea, I'll write expand the comment about `parentId` and add a link to Chrome DevTools docs to this

https://chromedevtools.github.io/devtools-protocol/tot/Runtime#type-StackTraceId

File: src/protocol/logs.rs (hunk 1 above)
Comment (RyuuGan) [reply src/protocol/logs.rs]: Basically I used the same type for timestamp as in `ResponseReceivedEventParams` from `network.rs`, but in chrome it means a little bit different, but represents still with the same `number` type.

Comment (atroche) [reply src/protocol/logs.rs]: I'm open to using chrono for these fields later on, but at the moment the other timestamps are all floats, so I don't think this should block the PR

File: src/protocol/logs.rs [hunk 3]
```diff
@@ -0,0 +1,212 @@
+pub mod events {
+    use serde::{Deserialize, Serialize};
+    use crate::protocol::runtime::methods::{StackTrace, RemoteObject};
+    use crate::protocol::types::{JsInt};
+
+    #[derive(Deserialize, Serialize, Debug, Clone, PartialEq, Eq)]
+    #[serde(rename_all = "lowercase")]
+    pub enum  LogEntrySource {
+        XML,
```
Comment (atroche) [inline src/protocol/logs.rs]: I'd prefer if these were in `UpperCamelCase` to match the style of enum variants in the rest of the project and also [these guidelines](https://rust-lang-nursery.github.io/api-guidelines/naming.html).

File: src/protocol/logs.rs [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+        entry: LogEntry,
+    }
+
+    #[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+    #[serde(rename_all = "camelCase")]
+    pub struct EntryAddedEvent {
+        pub params: EntryAddedParams,
+    }
+
+    #[test]
```
Comment (atroche) [inline src/protocol/logs.rs]: Thanks for adding tests! Really appreciate it.

File: src/protocol/runtime.rs [hunk 5]
```diff
@@ -25,19 +26,94 @@ pub mod methods {
...
+        DATAVIEW,
+    }
+
+    #[derive(Deserialize, Serialize, Debug, Clone, PartialEq)]
     #[serde(rename_all = "camelCase")]
     pub struct RemoteObject {
         #[serde(rename = "type")]
-        pub object_type: String,
-        pub subtype: Option<String>,
+        pub object_type: RemoteObjectType,
```
Comment (atroche) [inline src/protocol/runtime.rs]: Nice!

File: src/protocol/logs.rs (hunk 3 above)
Comment (atroche) [reply src/protocol/logs.rs]: heh, even though I happened to comment on the `XML` line which is one of the few that wouldn't need to change ;)

Comment (RyuuGan) [reply src/protocol/logs.rs]: Sure, I'll refactor the Enums, thanks for the link to guidelines.

**Comment by atroche** [comment]
Okay, still having CI problems on Mac (unrelated to your work). Feel free to merge when you've got it passing on Linux :)

File: src/protocol/logs.rs (hunk 3 above)
Comment (RyuuGan) [reply src/protocol/logs.rs]: Refactored to `UpperCamelCase`, also added forgotten `pub` in one of the enums. Also ran formatting with `cargo fmt`

**Pull request merged**